crates/primitives/src/transaction/eip1559.rs

```diff
@@ -24,7 +24,7 @@ pub struct TxEip1559 {
     /// this transaction. This is paid up-front, before any
     /// computation is done and may not be increased
     /// later; formally Tg.
-    pub gas_limit: u64,
+    pub gas_limit: u128,
```

Comment:
we can't do that here because the codec is slightly different for u128

instead we need to cast these fields when we implement the codec roundtrip